### PR TITLE
Update inference_api.md with pre-warming feature.

### PR DESCRIPTION
**Pull request by lukedeng95**
Update inference explainer with per warming feature.

### CODE REVIEW

File: inference-explainers/inference_api.md [hunk 1]
```diff
@@ -78,7 +79,19 @@ For example:
     {
       "model_path": "pcvr_v1/",
       "checksum": "dd94b3b08ea19b4240aa5e5f68ff0447b40e37ebdd06cc76fa1a2cf61143a10c",
-      "warm_up_batch_request_json": "..."
+      "warm_up_batch_request_json": "{
+          \"request\":[
```
Comment (steveganzy) [inline inference-explainers/inference_api.md:83]: Red indicates formatting errors, please resolve. You might wan to check how to escape json

File: inference-explainers/inference_api.md [hunk 2]
```diff
@@ -115,7 +128,7 @@ The listed steps are equivalent to the following bash command. This command work
 find <model_path> -type f -exec sha256sum {} \; | sort -k 2 | awk '{print $1}' | tr -d '\n' | sha256sum | awk '{print $1}'
 ```
 
-The optional `warm_up_batch_request_json` is a JSON batch request used to warm up the model before any traffic is served. This request is parsed and sent to the model to trigger initialization during the loading phase, reducing the latency impact of model lazy initialization during the first request. The format of this warm-up request uses the same format as the input to the runInference JavaScript function, which is described later in this document.
+The optional `warm_up_batch_request_json` is a JSON batch request used to warm up the model before serving traffic (NOTE: the request need to be escaped, please follow example format for reference). It follows the same format as the batch request for the runInference JavaScript function, which will be covered later in this document in the JavaScript API section. Ad techs can leverage model warm-up to reduce initial request latency and validate the model. This request is parsed and sent to the model during model registration. If the warm-up fails, the model is rejected.
```
Comment (akshaypundle) [inline inference-explainers/inference_api.md:131]: How can users escape their json? is there a 1 liner / shell command / something simple they can do for this?

File: inference-explainers/inference_api.md (hunk 1 above)
Comment (lukedeng95) [reply inference-explainers/inference_api.md:83]: This is due to we formate a string value for better readability, to resolve we will need to make it back to a non formatted string value which as "warm_up_batch_request_json": "{\"request\":[{\"model_path\":\"simple_model\",\"tensors\":[{\"tensor_name\":\"serving_default_input_1:0\",\"data_type\":\"FLOAT\",\"tensor_shape\":[1,1],\"tensor_content\":[\"0.0\"]}]}]}".  Wonder if we want to keep the readability?

File: inference-explainers/inference_api.md (hunk 2 above)
Comment (mihnjong-l) [reply inference-explainers/inference_api.md:131]: Can we make this change as a follow PR?

BTW, I think `jq` can do the work something like:

```echo $JSON | jq -R '.'```

File: inference-explainers/inference_api.md [hunk 3]
```diff
@@ -368,3 +381,5 @@ for (const response of inference_result.response) {
 [9]: https://github.com/tensorflow/tensorflow/blob/r2.14/tensorflow/python/framework/convert_to_constants.py#L1223
 [10]: https://github.com/privacysandbox/bidding-auction-servers/blob/release-4.5/services/inference_sidecar/modules/tensorflow_v2_14_0/tools/freeze_model.cc
 [11]: https://keras.io/
+[12]: https://github.com/lukedeng95
+
```
Comment (mihnjong-l) [inline inference-explainers/inference_api.md:385]: nit: no blank line.

**Review by mihnjong-l — CHANGES_REQUESTED** [review]
Please fix a typo in the commit message: "per warming" -> "pre-warming"

File: inference-explainers/inference_api.md (hunk 1 above)
Comment (lukedeng95) [reply inference-explainers/inference_api.md]: Edited to after formatted.

File: inference-explainers/inference_api.md (hunk 2 above)
Comment (lukedeng95) [reply inference-explainers/inference_api.md]: Thanks for the suggestion, Added in the explainer NOTE, with a general cmd: cat $JSON | tr -d '\n \t' | jq -R '.'

File: inference-explainers/inference_api.md [hunk 4]
```diff
@@ -115,7 +116,7 @@ The listed steps are equivalent to the following bash command. This command work
...
 ```
 
-The optional `warm_up_batch_request_json` is a JSON batch request used to warm up the model before any traffic is served. This request is parsed and sent to the model to trigger initialization during the loading phase, reducing the latency impact of model lazy initialization during the first request. The format of this warm-up request uses the same format as the input to the runInference JavaScript function, which is described later in this document.
+The optional `warm_up_batch_request_json` is a JSON batch request used to warm up the model before serving traffic (NOTE: the request need to be escaped, please follow example format for reference. Sample command for escaping pre-warming request:`cat $YOUR_WARM_UP_REQUEST_FILE | tr -d '\n \t' | jq -R '.'`). It follows the same format as the batch request for the `runInference` JavaScript function, which will be covered later in this document in the JavaScript API section. Ad techs can leverage model warm-up to reduce initial request latency and validate the model. This request is parsed and sent to the model during model registration. If the warm-up fails, the model is rejected.
```
Comment (mihnjong-l) [inline inference-explainers/inference_api.md:119]: How about:

NOTE: the request string must be escaped. Refer to the sample command to ensure correct formatting: `...`

Comment (akshaypundle) [inline inference-explainers/inference_api.md:119]: can we have the sample command on a different line (like the find above)?

Comment (lukedeng95) [reply inference-explainers/inference_api.md]: Sg, Split NOTE part out.

Comment (lukedeng95) [reply inference-explainers/inference_api.md]: Thanks for suggestion. Edited.

**Review by mihnjong-l — APPROVED** [review]

**Review by akshaypundle — APPROVED** [review]
